controllers/csiaddonsnode_controller.go

```diff
@@ -18,19 +18,31 @@ package controllers
 
 import (
 	"context"
+	"errors"
+	"fmt"
 
+	csiaddonsv1alpha1 "github.com/csi-addons/kubernetes-csi-addons/api/v1alpha1"
+	"github.com/csi-addons/kubernetes-csi-addons/internal/connection"
+	"github.com/csi-addons/kubernetes-csi-addons/internal/util"
+
+	"github.com/go-logr/logr"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
 
-	csiaddonsv1alpha1 "github.com/csi-addons/kubernetes-csi-addons/api/v1alpha1"
+var (
+	csiAddonsNodeFinalizer = csiaddonsv1alpha1.GroupVersion.Group + "/csiaddonsnode"
 )
 
 // CSIAddonsNodeReconciler reconciles a CSIAddonsNode object
 type CSIAddonsNodeReconciler struct {
 	client.Client
-	Scheme *runtime.Scheme
+	Scheme   *runtime.Scheme
+	ConnPool *connection.ConnectionPool
```

Comment:
all connections are stored in this CSIAddonsNodeReconciler struct how consumers will use it as it's not stored in some global variable?

Reply:
got this one as it's defined in main.go. Thanks @Rakshith-R

Reply:
> all connections are stored in this CSIAddonsNodeReconciler struct how consumers will use it as it's not stored in some global variable?

variable will be created in main.go and pointer will be passed to each controller as part of their struct(this change will be in respective controller prs).